src/Microsoft.NetCore.Analyzers/Core/Runtime/AvoidZeroLengthArrayAllocations.Fixer.cs

```diff
@@ -46,20 +48,50 @@ public sealed override FixAllProvider GetFixAllProvider()
             SemanticModel semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             SyntaxGenerator generator = editor.Generator;
 
-            SyntaxNode variableDecl = generator.GetDeclaration(nodeToFix, DeclarationKind.Variable);
-            SyntaxNode typeNode = generator.GetType(variableDecl);
+            var operation = semanticModel.GetOperationInternal(nodeToFix, cancellationToken);
+            var arrayCreationExpression = FindArrayCreation(operation);
```

Comment:
Please use explicit type when the type is not apparent from the right-hand side of an expression.